### PR TITLE
Feature - Resources admin dashboard

### PR DESCRIPTION
**Pull request by tfredh**

### CODE REVIEW

**Comment by tfredh** [comment]
Uhh, looks like my first PR's branch's commits were combined with this branch

**Comment by tfredh** [comment]
lemme know about any structuring problems irrelevant of size. I just kinda autopiloted for a couple hours in these 5 days XD

**Comment by tfredh** [comment]
had this ready to copy paste from my personal utility code dump

**Comment by tfredh** [comment]
good part of this was also from my code dump so it didn't take very long

**Comment by xsharonhe** [comment]
> Uhh, looks like my first PR's branch's commits were combined with this branch

next time: https://stackoverflow.com/questions/36168839/how-to-remove-commits-from-a-pull-request

File: components/HorizontalSeparator.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import styled from "styled-components";
+
+// function HorizontalSeparator() {
+//   return <></>;
+// }
```
Comment (xsharonhe) [inline components/HorizontalSeparator.jsx]: cleanup - remove this?

File: components/Input.jsx [hunk 2]
```diff
@@ -20,7 +20,8 @@ export const Input = ({
   />
 );
 
-const SInput = styled.input`
+// used when ref needs to be passed as prop
+export const SInput = styled.input`
```
Comment (xsharonhe) [inline components/Input.jsx:24]: I'm guessing this doesn't work with just the `Input`? I think if you want a `ref` as a prop you can make it on the original element and pass it as optional?

File: components/MDEditor.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import { useCallback, useState } from "react";
+import "easymde/dist/easymde.min.css";
+
+export default function MDEditor({
```
Comment (xsharonhe) [inline components/MDEditor.js]: Can you record a GIF demo for this?

File: components/resourcesAdmin/BlogTemplateRow.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    <BlogTemplateRowWrapper>
+      <BlogInformationContainer>
+        <Input disabled type="checkbox" />
+
+        <a className="title" href={`/resources/admin/project/blog/${blog.id}`}>
+          {blog.name}
+        </a>
+        <div className="modified-time-stamp">
+          Last changed {latestTimeAgo(blog.lastModified)} ago
+        </div>
```
Comment (xsharonhe) [inline components/resourcesAdmin/BlogTemplateRow.js]: I would be very careful about `className` - the purpose of styled components is to remove the need for class names so there isn't duplicates throughout our codebase. I think here might be fine since it is a very specific component but just keep this in mind

File: components/resourcesAdmin/BlogTemplateRow.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+export default function BlogTemplateRow({ blog }) {
+  return (
+    <BlogTemplateRowWrapper>
+      <BlogInformationContainer>
+        <Input disabled type="checkbox" />
+
+        <a className="title" href={`/resources/admin/project/blog/${blog.id}`}>
+          {blog.name}
+        </a>
```
Comment (xsharonhe) [inline components/resourcesAdmin/BlogTemplateRow.js]: like I think here you could get away with a nested link instead of `className`

File: components/resourcesAdmin/ProjectComponentStyles.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import styled from "styled-components";
+
+const HEADER_COLOUR = "#1e2530";
+const HEADER_HEIGHT = "6vh";
```
Comment (xsharonhe) [inline components/resourcesAdmin/ProjectComponentStyles.js]: I think it might be better to abstract this into a file in the `util` section instead of creating a new file for `resourcesAdmin` - maybe `utils/admin`?

Comment (xsharonhe) [reply components/resourcesAdmin/ProjectComponentStyles.js]: Are there anything in common with these components? I think they should be likely broken down into separate files because an external developer would have no idea which components are in which files.

Typically components go in the `components` file and then you can probably create an `admin` subfile from that. The const colours and height should go in the `theme/theme.js` file

File: contexts/AdminProvider.js [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+import { capitalizeEveryWord } from "../utils/methods/general";
+
+const Context = createContext();
+
+export function AdminProvider({ children }) {
+  const [adminUser, setAdminUser] = useState(null);
+
+  // attempt to log user in right after page load
+  useEffect(() => {
+    async function f() {
```
Comment (xsharonhe) [inline contexts/AdminProvider.js:11]: a little weird naming? if this is temporary probably fine

File: lib/envs.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export const ENVS = {
+  a: 1,
+};
```
Comment (xsharonhe) [inline lib/envs.js]: this is...? 😅

File: pages/api/admin/instant-log-in.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+import { validateToken } from "../../../middlewares/auth";
+
+export default async (req, res) => {
+  validateToken(req, res);
+
+  /**
+   * asdjkh
+   *
+   * stuff
+   */
```
Comment (xsharonhe) [inline pages/api/admin/instant-log-in.js]: remove comment here

**Review by xsharonhe — COMMENTED** [review]
I'll review more later but if you could show a demo of all this, it would be great. Kinda hard to follow all the files without this since I don't know how it is to be used as intended. Good work so far though :D

File: lib/envs.js (hunk 8 above)
Comment (tfredh) [reply lib/envs.js]: o XD I was making mistakes with env variables so I was going to just store them all here and check for attributes. after that I would import it if I ever needed env variables

**Comment by tfredh** [comment]
yeah np will do demo soon

**Comment by tfredh** [comment]
Small things
